test/wait.go

```diff
@@ -96,6 +96,25 @@ func WaitForTaskRunState(ctx context.Context, c *clients, name string, inState C
 	})
 }
 
+// WaitForV1TaskRunState polls the status of the TaskRun called name from client every
+// interval until inState returns `true` indicating it is done, returns an
+// error or timeout. desc will be used to name the metric that is emitted to
+// track how long it took for name to get into the state checked by inState.
+// TODO: needs to change all WaitForTaskRunState to v1 when switching the stored version
+func WaitForV1TaskRunState(ctx context.Context, c *clients, name string, inState ConditionAccessorFn, desc string) error {
```

Comment:
Instead of creating a separate function here, could the version be passed as an argument to `WaitForTaskRunState`? (same for PipelineRun)

Reply:
In most tests under ./test, `WaitForTaskRunState` and `WaitForPipelineRunState` are being used. And as put in #L103, they will all be swapped to `WaitForV1TaskRunState` and `WaitForV1PipelineRunState` once we change the stored version to v1. So I think this is just an intermediate state where we keep it and it shall not be necessary to add another param in this func's declaration?

Reply:
The timeline for removing these tests would be when we remove support for v1beta1, not when we swap the stored version to v1. I'd prefer to choose the option that keeps the codebase in a cleaner state the entire time. Removing an argument isn't really harder than removing a function and I don't think it's desirable to optimize for that compared to minimizing duplication.

Reply:
Thanks Lee. This makes perfect sense to me. Updated.